avm/src/vm/memory.rs

```diff
@@ -477,15 +501,20 @@ impl HandlerMemory {
     let s = hm.mem_addr; // The initial block that will be transferred (plus all following blocks)
     let s2 = self.mems.len(); // The new address of the initial block
     let offset = s2 - s; // Assuming it was made by `fork` this should be positive or zero
-    let (a, b) = hm.addr_to_idxs(CLOSURE_ARG_MEM_START); // The only address that can "escape"
-    hm.mems.drain(..s); // Remove the irrelevant memory blocks
-    self.mems.append(&mut hm.mems); // Append the relevant ones to the original HandlerMemory
-    // Set the return address on the original HandlerMemory to the acquired indexes, potentially
-    // offset if it is a pointer at new data
-    if a < std::usize::MAX && a >= s {
-      self.set_addr(CLOSURE_ARG_MEM_START, a + offset, b);
+    if hm.addr.1.len() > 0 {
+      let (a, b) = hm.addr_to_idxs(CLOSURE_ARG_MEM_START); // The only address that can "escape"
+      hm.mems.drain(..s); // Remove the irrelevant memory blocks
+      self.mems.append(&mut hm.mems); // Append the relevant ones to the original HandlerMemory
+      // Set the return address on the original HandlerMemory to the acquired indexes, potentially
+      // offset if it is a pointer at new data
+      if a < std::usize::MAX && a >= s {
+        self.set_addr(CLOSURE_ARG_MEM_START, a + offset, b);
+      } else {
+        self.set_addr(CLOSURE_ARG_MEM_START, a, b);
+      }
     } else {
```

Comment:
when exactly does this branch run?

Reply:
For any IO opcode that *doesn't* provide a return value. Eg `httpget` just writes the result into the `arg[2]` address. That's actually most io opcodes, but before we weren't using fork and join on opcodes directly.